python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -0,0 +1,302 @@
+        ----------
+        value : Boolean, mandatory
+            Value to set suppressDimSN to
+        '''
+        # if suppressDimSN is None:
+        #    self._suppressDimSN = True
+        # else:
+        self._suppressDimSN = suppressDimSN
+        return self._suppressDimSN
+
+    @property
+    def suppressHighzSN(self):
+        '''
+        Boolean to decide whether to output information corresponding to SN
+        at redshift above self.maxz
+        '''
+
+        return True
+
+    @property
+    def maxTimeSNVisible(self):
+        '''
+        The catalog will provide values for SN flux (even if zero according to
+        model) for peak \pm maxTimeVisibilityforSN
+        '''
+        return 100.
+
+    @property
+    def maxz(self):
+        return 1.2
+
+    @astropy.utils.lazyproperty
+    def photometricparameters(self, expTime=15., nexp=2):
+        lsstPhotometricParameters = PhotometricParameters(exptime=expTime,
+                                                          nexp=nexp)
+        return lsstPhotometricParameters
+
+    @astropy.utils.lazyproperty
+    def lsstBandpassDict(self):
+        return BandpassDict.loadTotalBandpassesFromFiles()
+
+    @astropy.utils.lazyproperty
+    def observedIndices(self):
+        bandPassNames = self.obs_metadata.bandpass
+        return [self.lsstBandpassDict.keys().index(x) for x in bandPassNames]
+
+    @astropy.utils.lazyproperty
+    def lsstpbase(self):
+        pbase = PhotometryBase()
+        return pbase
+
+    def get_snid(self):
+        # Not necessarily unique if the same galaxy hosts two SN
+        # Use refIdCol to access the relevant id column of the dbobj
+        # Should revert to galTileID for galaxyTiled catalogDBObj and
+        # id for galaxyObj catalogDBObj
+        # (email from Scott)
+        return self.column_by_name(self.refIdCol)
+
+    @property
+    def numobjs(self):
+        return len(self.column_by_name('id'))
+
+    @compound('snra', 'sndec', 'z', 'vra', 'vdec', 'vr')
+    def get_angularCoordinates(self):
+        '''
+        Obtain the coordinates and velocity of the SN from the host galaxy
+
+        Returns
+        -------
+        `np.ndarray` of coordinara, dec, z, vra, vdec, and vr
+
+        '''
+        hostra, hostdec, hostz = self.column_by_name('raJ2000'),\
+            self.column_by_name('decJ2000'),\
+            self.column_by_name('redshift')
+        snra, sndec, snz, snvra, snvdec, snvr = self.SNCoordinatesFromHost(
+            hostra, hostdec, hostz)
+
+        return ([snra, sndec, snz, snvra, sndec, snvr])
+
+    @compound('c', 'x1', 'x0', 't0')
+    def get_snparams(self):
+        hostz, hostid, hostmu = self.column_by_name('redshift'),\
+            self.column_by_name('snid'),\
+            self.column_by_name('cosmologicalDistanceModulus')
+
+        vals = self.SNparamDistfromHost(hostz, hostid, hostmu)
+
+        return (vals[:, 0], vals[:, 1], vals[:, 2], vals[:, 3])
+
+    def get_SNsed(self):
```

Comment:
defining this as a get_ method means that it is possible for a user to try to write it out into a catalog.  I do not know what will happen if they try to write SEDs out to an InstanceCatalog.  Can you change the name of this method to something like 'loadSEDs' so that there is no risk of SNsed being added to column_outputs?